matRad_calcPhotonDoseVmc.m

```diff
@@ -275,7 +287,7 @@
                     if calcDoseDirect
                         if isfield(stf(i).ray(j),'weight')
                             % score physical dose
-                            dij.physicalDose{1}(:,1) = dij.physicalDose{1}(:,1) + stf(dij.beamNum(readCounter)).ray(dij.rayNum(readCounter)).weight * doseTmpContainer{1,1};
+                            dij.physicalDose{1}(:,i) = dij.physicalDose{1}(:,i) + stf(i).ray(j).weight * doseTmpContainer{1,1};
                         else
```

Comment:
Any particular reason why you are not using 

stf(dij.beamNum(readCounter)).ray(dij.rayNum(readCounter)).weight

anymore? Did you double check everything with dirty multithreading activated? I figure this is wrong?!